models/user.js

```diff
@@ -1,10 +1,148 @@
 const mongoose = require('mongoose');
+const bcrypt = require('bcrypt');
 
-const User = new mongoose.Schema({
-  name: String,
-  hash: String,
-  salt: String,
-  cookie: String,
+const {
+	Schema,
+} = mongoose;
+
+const fields = {
+	firstname: {
+		type: String,
+		required: true,
+	},
+	lastname: {
+		type: String,
+		required: true,
+	},
+	email: {
+		type: String,
+		trim: true,
+		lowercase: true,
+		required: true,
+	},
+	password: {
+		type: String,
+		required: true,
+	},
+};
+
+const user = new Schema(fields, {
```

Comment:
schema name should be capital to avoid confusion between instances and the schema